### ISSUE TITLE
Config htaccess.RewriteBase makes admin datadir check fail

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/server#1364]
### Steps to reproduce
1. Install
2. Set config `'htaccess.RewriteBase' => '/',`
3. Update the htaccess `./occ maintenance:update:htaccess`
4. Open admin page
### Expected behaviour

No warning
### Actual behaviour

> Your data directory and your files are probably accessible from the Internet. The .htaccess file is not working. We strongly suggest that you configure your web server in a way that the data directory is no longer accessible or you move the data directory outside the web server document root.

The problem is, that the check returns a 302, therefor does not match any more:
https://github.com/nextcloud/server/blob/44f3fcc187ed01a856dbd0ebdb2f2a090072d528/core/js/setupchecks.js#L201-L201

Not sure if adding 302 there is "save". Therefor I created the issue.
Maybe @LukasReschke knows more.

Reported by ThinkPrivacy in IRC

**Comment by mageroya** [comment]
I cannot judge if it is safe to add 302 to the list, but I can tell you it **does not fix** the issue, I tried locally. 

changed JS code to 
```
if (xhr.status !== 403 && xhr.status !== 302 && xhr.status !== 307 && xhr.status !== 301 && xhr.responseText !== '') {
  messages.push({
  msg: t('core', 'xhr.status == ' + xhr.status),
  type: OC.SetupChecks.MESSAGE_TYPE_ERROR
});
```
and the admin page shows me this message:
```
xhr.status == 200
```
Apparently AJAX immediatly fires off another GET with the new location when receiving a 302, since the 302 does not even make it into this handeling function.

**Comment by LukasReschke** [comment]
Let's be a little bit more error resistant and do the following:

1. Change the text in https://github.com/nextcloud/server/blob/3431591f34750fd9e7a085d7e041c1e762c86c52/lib/private/legacy/util.php#L1135-L1159 to remove the "This file can be safely removed."
2. Check xhr.responseText for the value of the text file

**Comment by MorrisJobke** [comment]
Fix is in #2513
